2025-01-06 09:03:46,240: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2025-01-06 09:03:46,248: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2025-01-06 09:03:46[0m
2025-01-06 09:03:46,256: CcxtCacheDB: INFO: download ranges :
+--------+------+
| from   | to   |
|--------+------|
+--------+------+
2025-01-06 09:03:46,262: CcxtCacheDB: INFO: cache ranges:
+----+----------------------------+----------------------------+
|    | from                       | to                         |
|----+----------------------------+----------------------------|
|  0 | 2023-02-04T00:00:00.000000 | 2024-09-24T00:00:00.000000 |
+----+----------------------------+----------------------------+
2025-01-06 09:03:48,422: httpx: INFO: HTTP Request: POST http://127.0.0.1:11434/api/generate "HTTP/1.1 200 OK"
2025-01-06 09:03:48,752: asyncio: INFO: [MLTrader] [32mNew order was created: market order of | 1.0 BTC/USD buy | at price $46428.0 of class bracket with status new[0m
2025-01-06 09:03:48,761: asyncio: INFO: [MLTrader] [34mTrading iteration ended at 2025-01-06 09:03:48[0m
2025-01-06 09:03:48,761: asyncio: INFO: [MLTrader] Getting historical prices for (BTC, USD), 1 bars, 
2025-01-06 09:03:48,762: root: INFO: market order of | 1.0 BTC/USD buy | at price $46428.0 of class bracket with status new was sent to broker backtesting
2025-01-06 09:03:48,762: root: INFO: market order of | 1.0 BTC/USD buy | at price $46428.0 of class bracket with status new was sent to broker backtesting
2025-01-06 09:03:48,763: asyncio: INFO: [MLTrader] [32mOrder was filled: market order of | 1.0 BTC/USD buy | at price $37719.0 of class bracket with status fill[0m
2025-01-06 09:03:48,766: asyncio: INFO: [MLTrader] [34mSleeping for 86400 seconds[0m
2025-01-06 09:03:48,766: root: INFO: Current backtesting datetime 2023-12-02 00:00:00-05:00
2025-01-06 09:03:48,767: CcxtCacheDB: INFO: download ranges :
+---------------------+----------------------------+
| from                | to                         |
|---------------------+----------------------------|
| 2023-02-04 00:00:00 | 2023-12-31 23:59:59.999999 |
+---------------------+----------------------------+
2025-01-06 09:03:48,767: root: ERROR: A request for market data for USD/USDT was submitted. The market for that pair does not exist
2025-01-06 09:03:48,767: asyncio: ERROR: [MLTrader] 'NoneType' object has no attribute 'set_index'
2025-01-06 09:03:48,770: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 923, in run
    self._run_trading_session()
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 871, in _run_trading_session
    self._on_trading_iteration()
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 284, in func_output
    self.strategy._update_portfolio_value()
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/strategies/_strategy.py", line 449, in _update_portfolio_value
    prices = self.broker.data_source.get_last_prices(assets)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/data_sources/data_source.py", line 348, in get_last_prices
    result[asset] = self.get_last_price(asset, quote=quote, exchange=exchange)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/data_sources/ccxt_backtesting_data.py", line 242, in get_last_price
    bars = self.get_historical_prices(asset, 1, timestep=timestep, quote=quote, timeshift=None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/data_sources/ccxt_backtesting_data.py", line 174, in get_historical_prices
    response = self._pull_source_symbol_bars(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/data_sources/ccxt_backtesting_data.py", line 109, in _pull_source_symbol_bars
    data = self._pull_source_bars([asset],length,timestep,timeshift,quote,include_after_hours)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/data_sources/ccxt_backtesting_data.py", line 154, in _pull_source_bars
    data = self.cache_db.download_ohlcv(symbol,parsed_timestep,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/tools/ccxt_data_store.py", line 189, in download_ohlcv
    df = self._fill_missing_data(df, timeframe)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/devin/lib/python3.11/site-packages/lumibot/tools/ccxt_data_store.py", line 447, in _fill_missing_data
    df.set_index("datetime", inplace=True)
    ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'set_index'

2025-01-06 09:03:48,770: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2025-01-06 09:03:49,726: root: INFO: Backtesting finished
2025-01-06 09:03:50,937: root: ERROR: Not enough data to create a tearsheet, at least 2 days of data are required. Skipping
2025-01-06 09:03:50,937: backtest_stats: INFO: Backtest took 0:00:04.697867 for a speed of 0.000
